### PR TITLE
AC-424:Do not require login location if no locations are configured.

### PR DESCRIPTION
**Pull request by anonymous-ME**
The necessary changes as pointed out by @ Eric Henziger

### CODE REVIEW

File: openmrs-client/src/main/java/org/openmrs/mobile/listeners/watcher/LoginValidatorWatcher.java [hunk 1]
```diff
@@ -143,7 +143,14 @@ public void setLocationErrorOccurred(boolean locationErrorOccurred) {
         this.locationErrorOccurred = locationErrorOccurred;
     }
 
+
     public boolean isLocationErrorOccurred() {
         return locationErrorOccurred;
     }
+
+    public boolean validateSpinner(Spinner spinner) {
```
Comment (henziger) [inline openmrs-client/src/main/java/org/openmrs/mobile/listeners/watcher/LoginValidatorWatcher.java]: This method can be private for now I think. And perhaps also move this method to directly after validateNotEmpty to keep similar methods close to each other. :)

**Comment by dkayiwa** [comment]
@anonymous-ME can you include the pull request url as advised at? https://wiki.openmrs.org/display/docs/Pull+Request+Tips

**Comment by dkayiwa** [comment]
@anonymous-ME did you see the above comment? 😊

File: openmrs-client/src/main/java/org/openmrs/mobile/activities/login/LoginFragment.java [hunk 2]
```diff
@@ -143,10 +144,13 @@ public void onFocusChange(View view, boolean b) {
         mLoginButton.setOnClickListener(new View.OnClickListener() {
             @Override
             public void onClick(View v) {
-                mPresenter.login(mUsername.getText().toString(),
-                        mPassword.getText().toString(),
-                        mUrl.getText().toString(),
-                        initialUrl);
+                if (LoginValidatorWatcher.isAllDataValid())
```
Comment (shivtej1505) [inline openmrs-client/src/main/java/org/openmrs/mobile/activities/login/LoginFragment.java]: @anonymous-ME, why can't we just use instance variable here?

File: openmrs-client/src/main/java/org/openmrs/mobile/activities/login/LoginFragment.java [hunk 3]
```diff
@@ -143,10 +144,13 @@ public void onFocusChange(View view, boolean b) {
...
+                    mPresenter.login(mUsername.getText().toString(),
+                            mPassword.getText().toString(),
+                            mUrl.getText().toString(),
+                            initialUrl);
+                else
+                    Toast.makeText(getContext(),"Please enter valid data to login.",Toast.LENGTH_LONG).show();
```
Comment (shivtej1505) [inline openmrs-client/src/main/java/org/openmrs/mobile/activities/login/LoginFragment.java]: Could you move this hard-coded string to `strings.xml`?

File: openmrs-client/src/main/java/org/openmrs/mobile/listeners/watcher/LoginValidatorWatcher.java [hunk 4]
```diff
@@ -114,9 +114,9 @@ public void onNothingSelected(AdapterView<?> adapterView) {
         // This method is intentionally empty
     }
 
-    private boolean isAllDataValid() {
+    public static boolean isAllDataValid() {
```
Comment (shivtej1505) [inline openmrs-client/src/main/java/org/openmrs/mobile/listeners/watcher/LoginValidatorWatcher.java]: Why does this method need to be static?

**Comment by dkayiwa** [comment]
@anonymous-ME can you include a link to the ticket as advised at? https://wiki.openmrs.org/display/docs/Pull+Request+Tips

**Comment by dkayiwa** [comment]
@anonymous-ME did you see the above comment?

**Comment by dkayiwa** [comment]
@anonymous-ME ping me to reopen when you resume working on this

**Pull request closed without merging**